backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewToolService.java

```diff
@@ -268,4 +273,53 @@ public Element dropRequirementSubjectFromDiagram(Element droppedElement, Node dr
+            Map<org.eclipse.sirius.components.view.diagram.NodeDescription, NodeDescription> convertedNodes) {
+        if (droppedElement instanceof ConstraintUsage droppedConstraint) {
+            if (targetElement instanceof RequirementUsage requirementUsage) {
+                this.moveContraintInRequirementConstraintCompartment(droppedConstraint, requirementUsage, RequirementConstraintKind.ASSUMPTION);
+                this.createView(droppedElement, editingContext, diagramContext, targetNode, convertedNodes, NodeContainmentKind.CHILD_NODE);
+                diagramContext.getViewDeletionRequests().add(ViewDeletionRequest.newViewDeletionRequest().elementId(droppedNode.getId()).build());
+            }
+        }
+        return droppedElement;
+    }
+
+    public Element dropElementFromDiagramInRequirementRequireConstraintCompartment(Element droppedElement, Node droppedNode, Element targetElement, Node targetNode, IEditingContext editingContext, IDiagramContext diagramContext,
+            Map<org.eclipse.sirius.components.view.diagram.NodeDescription, NodeDescription> convertedNodes) {
+        if (droppedElement instanceof ConstraintUsage droppedConstraint) {
+            if (targetElement instanceof RequirementUsage requirementUsage) {
+                this.moveContraintInRequirementConstraintCompartment(droppedConstraint, requirementUsage, RequirementConstraintKind.REQUIREMENT);
+                this.createView(droppedElement, editingContext, diagramContext, targetNode, convertedNodes, NodeContainmentKind.CHILD_NODE);
+                diagramContext.getViewDeletionRequests().add(ViewDeletionRequest.newViewDeletionRequest().elementId(droppedNode.getId()).build());
+            }
+        }
+        return droppedElement;
+    }
+
+    private void moveContraintInRequirementConstraintCompartment(ConstraintUsage droppedConstraint, RequirementUsage requirementUsage, RequirementConstraintKind kind) {
+        var oldMembership = droppedConstraint.eContainer();
+        var membership = SysmlFactory.eINSTANCE.createRequirementConstraintMembership();
+        membership.getOwnedRelatedElement().add(droppedConstraint);
+        membership.setKind(kind);
+        requirementUsage.getOwnedRelationship().add(membership);
+        EcoreUtil.delete(oldMembership);
+    }
+
+    public Element dropElementFromDiagramInConstraintCompartment(Element droppedElement, Node droppedNode, Element targetElement, Node targetNode, IEditingContext editingContext, IDiagramContext diagramContext,
+            Map<org.eclipse.sirius.components.view.diagram.NodeDescription, NodeDescription> convertedNodes) {
+        if (droppedElement instanceof ConstraintUsage droppedConstraint) {
+            if (targetElement instanceof ConstraintUsage || targetElement instanceof ConstraintDefinition) {
+                var oldMembership = droppedConstraint.eContainer();
+                var membership = SysmlFactory.eINSTANCE.createFeatureMembership();
```

Comment:
Why create a new membership and delete the existing one?
Could you just move the existing membership instead?

Reply:
It's the same explanation as before.